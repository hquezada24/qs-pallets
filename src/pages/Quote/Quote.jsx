import styles from "./Styles.module.css";

const Quote = () => {
  return <div className={styles.quote}></div>;
};

export { Quote };
