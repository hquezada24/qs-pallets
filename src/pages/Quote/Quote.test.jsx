// Quote.test.jsx
// @vitest-environment jsdom
import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Quote } from "./Quote";

// Mock ScrollRestoration from react-router-dom
vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    ScrollRestoration: () => null,
  };
});

// Mock quote form component
vi.mock("../../components/forms/QuoteForm", () => ({
  default: () => <div data-testid="quote-form">QuoteForm</div>,
}));

// Wrapper component for Router
const renderWithRouter = (component) => {
  return render(<BrowserRouter>{component}</BrowserRouter>);
};

describe("Quote component", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("renders quote form component", () => {
    const { container } = renderWithRouter(<Quote />);

    expect(screen.getByTestId("quote-form")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot", () => {
    const { container } = renderWithRouter(<Quote />);
    expect(container).toMatchSnapshot();
  });
});
