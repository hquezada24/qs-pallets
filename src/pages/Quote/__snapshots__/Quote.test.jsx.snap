// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Quote component > matches snapshot 1`] = `
<div>
  <div
    class="_quote_b1fc99"
    id="quote"
  >
    <div
      data-testid="quote-form"
    >
      QuoteForm
    </div>
  </div>
</div>
`;

exports[`Quote component > renders quote form component 1`] = `
<div>
  <div
    class="_quote_b1fc99"
    id="quote"
  >
    <div
      data-testid="quote-form"
    >
      QuoteForm
    </div>
  </div>
</div>
`;
